scripts/creation-portal-migration/template/generate/gProgram.js

```diff
@@ -0,0 +1,547 @@
+      // Update the database if contributor update fails
+      await updateSolutionDb(query, solution, migratedCount);
+      return;
+    }
+
+    logger.info(
+      `Sourcing Program added contributor for solution_id: ${solution?._id} === ${programId}`
+    );
+
+    // update the query with isContributorAdded
+    query = {
+      ...query,
+      [constants.MIGRATION_REFERENCE.IS_CONTRIBUTOR_ADDED]: true,
+    };
+  } else {
+    // increment program contributor added success count
+    migratedCount.success.program.existing.contributor++;
+  }
+
+  if (!solution?.migrationReference?.isContributorAccepted) {
+    const accept_contributor = {
+      program_id: programId,
+      user_id:
+        userData?.srcOrgAdmin?.mappedUserId ||
+        process.env.DEFAULT_CONTRIBUTOR_ORG_ADMIN_ID,
+      status: "Approved",
+      updatedby: userId,
+    };
+    // call the api to accept contributor
+    const updateNomination = await updateContributorToProgram(
+      accept_contributor
+    ).catch((err) => {
+      logger.error(`Error while accepting nomination to the program for solution_id: ${
+        solution?._id
+      } Error:
+        ${JSON.stringify(err?.response?.data)}`);
+      // increment program contributor accepted failed count and store the id
+      migratedCount.failed.program.accepted.count++;
+      if (!migratedCount.failed.program.accepted.ids.includes(id)) {
+        migratedCount.failed.program.accepted.ids.push(id);
+      }
+    });
+
+    if (!updateNomination) {
+      // Update the db with only migrationReference with sourcingProgramId, isSourceProgramUpdated, isSourceProgramPublished, isNominated,
+      // isContributorAdded, if failed to accept contributor
+      await updateSolutionDb(query, solution, migratedCount);
+      return;
+    }
+    logger.info(
+      `Sourcing Program accepted nomination for solution_id: ${solution?._id} === ${programId}`
+    );
+    // update the query with isContributorAccepted
+    query = {
+      ...query,
+      [constants.MIGRATION_REFERENCE.IS_CONTRIBUTOR_ACCEPTED]: true,
+    };
+  } else {
+    // increment program contributor accepted success count
+    migratedCount.success.program.existing.accepted++;
+  }
+
+  if (!isEmpty(query)) {
+    // Update the db with only migrationReference with sourcingProgramId, isSourceProgramUpdated, isSourceProgramPublished, isNominated,
+    // isContributorAdded, isContributorAccepted if there is no failure
+    await updateSolutionDb(query, solution, migratedCount);
+  }
+  return { programId, contributor: userData?.srcOrgAdmin };
+};
+
+/**
+ * To update the program migration details in mongodb
+ * @method
+ * @name updateSolutionDb
+ * @param {Object} query - fields to update in mongodb.
+ * @param {Object} solution - solution Data.
+ * @param {Object} migratedCount - migratedCount to increment the count.
+ */
+const updateSolutionDb = async (query, solution, migratedCount) => {
+  const res = await updateSolutionById({
```

Comment:
@jayesh9747 use try catch and optimize the code with the help of AI
sample 
const updateSolutionDb = async (query, solution, migratedCount) => {
  try {
    // Update the solution by ID
    await updateSolutionById({
      id: solution?._id?.toString(),
      query: { ...query },
    });
  } catch (err) {
    logger.error(
      `Error while updating program in solutions collection: ${solution?._id} Error: ${err}`
    );
    return; // Early exit on error
  }

  // Define a mapping between query keys and migratedCount fields
  const migrationMapping = {
    [constants.MIGRATION_REFERENCE.SOURCING_PROGRAM_ID]:
      "migrated",
    [constants.MIGRATION_REFERENCE.IS_SRC_PROGRAM_UPDATED]:
      "updated",
    [constants.MIGRATION_REFERENCE.IS_SRC_PROGRAM_PUBLISHED]:
      "published",
    [constants.MIGRATION_REFERENCE.IS_NOMINATED]:
      "nominated",
    [constants.MIGRATION_REFERENCE.IS_CONTRIBUTOR_ADDED]:
      "contributor",
    [constants.MIGRATION_REFERENCE.IS_CONTRIBUTOR_ACCEPTED]:
      "accepted",
  };

  // Loop through the query keys and increment the corresponding counts
  Object.keys(query).forEach((key) => {
    if (migrationMapping[key]) {
      migratedCount.success.program.current[migrationMapping[key]]++;
    }
  });
};